src/InteractiveView.js

```diff
@@ -0,0 +1,14 @@
+import View from './View';
+
+const INTERACTIVE_NAVIGATION_START_MARK_NAME = 'transitionStart';
+
+export default class InteractiveView extends View {
+	startMarkName = null;
+
+	startTransition() {
```

Comment:
P.S. our API creates gap between starting new view with UXCapture.startView() and starting transition with UXCapture.startTransition() - during this time no marks should be fired (as interactive transition start mark hasn't been fired yet and is null). This is not great as we don't really control when they fire (e.g. onload events for images, for example). We can also keep view "un-attached" until it's started so all marks during this gap time are accounted towards previous view. Thoughts?

@sergeychernyshev @mmcgahan Calling out this from the description so we can make a thread on it.

Reply:
Link to the API spec: https://meetup.atlassian.net/wiki/spaces/WEG/pages/756319487/UX+Capture+API+Spec

Reply:
From my understanding of the API spec we reviewed before, `startView` and `startTransition` was intended to be used with a single `View` class, not separate `PageView` and `InteractiveView` classes.

With a single `View` class, you call `startView`, then any marks and measures for the page view are being recorded. Then you call `startTransition` when you want to begin an interactive view. `startView` is only ever called once, and `startTransition` udates the view.

Mike, feel free to add more context or correct me.

Reply:
Yeah - I elaborated a bit more in a separate comment, but I think we can make things a bit more clear by keeping `startTransition` logic out of `View` (or `View` subclasses).

both `UXCapture.startView` and `UXCapture.startTransition` will be called multiple times in any given session, but I don't think we have to worry about any 'gaps'. The expected application beahvior is:

1. see a request for a new route (through React-Router or `history.push`)
2. fire `UXCapture.startTransition`
3. Application renders a new view container for the route
4. View container fires `UXCapture.startView`
5. Repeat

This sequence should ensure that any marks corresponding to the view in (4) won't fire until the `View` instance has been created/started

Reply:
It looks like we under-documented the relationship between `UXCapture.startTransition` and `UXCapture.startView`.

I was under the impression that views are created using `UXCapture.startView` in advance and `UXCapture.startTransition` what makes them actually kick-in.

And you designed it for `UXCapture.startTransition` fire first just to mark the fact that user interacted with the app and next view is about to come, then `UXCapture.startView` firing and defining the zones/marks to be expected.

Now when you explained it, I understand why that makes sense for React apps where view definition happens later in the flow, but before components are updated.

I probably understood that already when we discussed the API and just forgot with all the commotion and didn't have much documentation to convince me otherwise ;)

Will rewrite the code and probably create a separate ticket to create a sequence diagram.

Reply:
sounds good - I'll definitely need to validate my assumptions about the current router lifecycle once we get into the React implementation, but we should be able to find the right hooks to tie in to

Reply:
Yeah, this makes sense. However, what's confusing is that the current API documentation spec says that `startView` will be called only once. We should update the spec along with this ticket.